lib/matplotlib/backend_tools.py

```diff
@@ -111,6 +117,23 @@ def destroy(self):
         """
         pass
 
+    def state_changed_event(self, sender, canvasevent, data):
```

Comment:
At the same time, the `tool_trigger_event` in `NavigationBase` should be renamed to `trigger_tool` or something like that, it is not an event firer, the event is fired by the tool not by navigation